Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -190,6 +190,9 @@ async def generate(self, request):
                     response_sender.send(self.create_response(output))
                 else:
                     last_output = output
+                if response_sender.is_cancelled():
+                    await self.llm_engine.abort(request_id)
```

Comment:
Why do we need await here? We don't seem to be grabbing the result.

Also, is this the right place to check if the response_sender is cancelled?

Reply:
> Why do we need await here? We don't seem to be grabbing the result.

`llm_engine.abort()`: https://github.com/vllm-project/vllm/blob/main/vllm/engine/async_llm_engine.py#L438

We want the abort to stop `async for output in self.llm_engine.generate()` loop from generating more output.

The `break` underneath is a failsafe in-case more output is generated, so the `llm_engine.abort()` is called at most once, otherwise it could raise an exception. It is also safe to break in such case because the request is cancelled and the results are no longer anticipated by the user.

> Also, is this the right place to check if the response_sender is cancelled?

Yes, because each output is the aggregated words that forms the response sentence. For example:
```
loop 0: 
loop 1: I
loop 2: I'm
loop 3: I'm sorry
loop 4: I'm sorry,
loop 5: I'm sorry, but
loop 6: I'm sorry, but I
loop 7: I'm sorry, but I'm
loop 8: I'm sorry, but I'm not
...
```
If the user no longer anticipates for the response (a.k.a. cancelled), it is safe to abort the generation and free up resources for other requests.